### DISCUSSION TITLE
To process all mails present in a folder

### DISCUSSION THREAD

**Discussion by aadeshd** [ecederstrand/exchangelib#1072] [Q&A]
I want to iterate through all mails in specified mail box folder and do some operation on the same.
`pathToFolder = account.root/ pathtospecifedfolder`
`msgs = pathToFolder.all()`

and to count the number of mails present in the folder, I am using a for loop as follows,
`c = 0`
`for i in msgs:`
    `    c = c +1`

This loop is giving me a correct count of number of mails present in the mailbox i.e 150.

Though while iterating again for processing each mail, it is extracting only 100 mails without any warning or error and execution stops after 100 mails (without any error msg).

`for item in msgs.iterator():`
`     do some operation on item`

I found out the default chunk_size value is 100, so I changed it to 300 manually from file "services/common.py".
then all mails present in mail box i.e. 150 got processed but I am getting an warning as 
"chunk size 300 is dangerously high, ResolveNames supports returning at most 100 candidates for a lookup"

I also tried changing this for loop as,
`for c in chunkify (msgs.iterator(),10):`
`      for item in c:`
            `do some operation on item`
still only 100 mails got extracted

I am new to this, can someone please help me resolve this issue and extract all mails present in the folder without any warning.

**Comment by ecederstrand** [comment]
Which version of exchangelib is this?

**Reply by aadeshd** [reply]
4.5.1

**Reply by aadeshd** [reply]
I am using another for loop just to get datetime of first item,
`for i in msgs:`
`firstItemTime = i.datetime_received`
`break`

Here is another code snippet I tried,

`for item in msgs[:c]:` # c is the correct count I got from first loop
`do some operation on item`

still only 100 mails got processed.

**Reply by ecederstrand** [reply]
You should avoid changing items that you are currently iterating over. It's like changing a list in Python while you are iterating over the items in the list. It can cause all sorts of problems.
